src/gluonnlp/data/batchify/embedding.py

```diff
@@ -135,7 +135,8 @@ def __iter__(self):
         if isinstance(self._sentences[0][0], str):
             sentences = [np.asarray(s, dtype='O') for s in self._sentences]
         else:
-            sentences = [np.asarray(s) for s in self._sentences]
+            dtype = type(self._sentences[0][0])
```

Comment:
What if the first sentence is empty?

Reply:
Added itertools.chain to workaround empty first sentence